narwhals/_duckdb/expr.py

```diff
@@ -447,6 +512,61 @@ def round(self: Self, decimals: int) -> Self:
             lambda _input: FunctionExpression("round", _input, lit(decimals)), "round"
         )
 
+    def cum_sum(self, *, reverse: bool) -> Self:
+        def func(
+            _input: duckdb.Expression,
+            partition_by: Sequence[str],
+            order_by: Sequence[str],
+        ) -> duckdb.Expression:
+            if reverse:
+                order_by_sql = "order by " + ", ".join(
+                    f'"{x}" desc nulls last' for x in order_by
+                )
+            else:
+                order_by_sql = "order by " + ", ".join(
+                    f'"{x}" asc nulls first' for x in order_by
+                )
+            if partition_by:
+                partition_by_sql = "partition by " + ",".join(
+                    f'"{x}"' for x in partition_by
+                )
+            else:
+                partition_by_sql = ""
+            sql = f"sum ({_input}) over ({partition_by_sql} {order_by_sql} rows between unbounded preceding and current row)"
+            return duckdb.SQLExpression(sql)
```

Comment:
I can see a lot of re-usability if everything here for #2174 - only "word" to input is "sum" and other operations 🙌🏼